packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -72,32 +68,10 @@ export default class RouterService extends Service {
       return router;
     }
     const owner = getOwner(this) as Owner;
-    router = owner.lookup('router:main') as EmberRouter;
-    router.setupRouter();
+    router = getRouter(owner) as EmberRouter | null;
```

Comment:
I think the original code here is fine, right? As long as we remove the DI injection (which I think was already done?) and remove the `constructor` here completely, I think we should be good to go.

Reply:
> Since we only need to forward the events when someone calls routerService.on('routeWillChange'), why don't we implement on and setup the wiring there?

That was actually one solution I initially looked at.  My thought is that, since these events are ultimately emanating from `router:main`, it's easier to reason about to have it be responsible for broadcasting events to where they need to go, rather than have `service:router` be responsible for looking up the router and forwarding events from that to itself and counting on `router:main` being in existence.  Having `service:router` lookup `router:main` when `on` is called, setup listening to events from `router:main`, seems a bit more complicated than just calling `trigger` on `service:router` at the point where we're already emitting these events.  Since `service:router` is really like a safe proxy around `router:main`, I think it makes sense to keep the service as passive as possible, including any hooking up of events.

I think it's not necessarily bad to override `on` in `service:router` to get this job done; philosophically, I believe it's a touch clearer not to do it that way.  A possible drawback to the approach in this PR currently is that `service:router` is always looked up when `route{Will|Did}Change` gets called, but `lookup` _seems_ fairly lightweight.

Reply:
> Since we only need to forward the events when someone calls routerService.on('routeWillChange'), why don't we implement on and setup the wiring there?

That was actually one solution I initially looked at.  My thought is that, since these events are ultimately emanating from `router:main`, it's easier to reason about to have it be responsible for broadcasting events to where they need to go, rather than have `service:router` be responsible for looking up the router and forwarding events from that to itself and counting on `router:main` being in existence.  Having `service:router` lookup `router:main` when `on` is called, setup listening to events from `router:main`, seems a bit more complicated than just calling `trigger` on `service:router` at the point where we're already emitting these events.  Since `service:router` is really like a safe proxy around `router:main`, I think it makes sense to keep the service as passive as possible, including any hooking up of events.

I don't think it's bad to override `on` in `service:router` to get this job done; philosophically, I believe it's a touch clearer not to do it that way.  The one advantage to it I see is that setting up event listening would only happen once, whereas looking up `service:router` and calling `trigger` on it would happen every time `route{Will|Did}Change` happens, although it _seems_ like `lookup` is pretty lightweight.

Reply:
Oh and touching on the `setupRouter`, I though we may have discussed that this might be unnecessary/duplicated.  I could be remembering wrong as well.  Happy to bring that back if you would rather keep it here.   

https://github.com/emberjs/ember.js/blob/8c85b2b52f304d7ca3987f97d9d2fbf18bd40d37/packages/%40ember/-internals/routing/lib/services/routing.ts#L33

Reply:
@Ravenstine @snewcomer Hi think keeping the call to router.setupRouter() in the service is ok here. I even think the whole setupRouter() thing should be removed, and the logic should be put in the router constructor.

Reply:
@sly7-7 we can probably remove the lazy initialization code from the router service ya, but I think `router.setupRouter()` itself has to stick around...